File: 03a.pl
#! /usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

my %PRIORITY = map { $_    => ord($_) - ord '`',
                     uc $_ =>  ord($_) - 70
               } 'a' .. 'z';

my $sum;
while (<>) {
    chomp;
    my $length = length;
    my $compartment = substr $_, 0, $length / 2, "";
    my %part;
    @part{split //} = ();
    my $common = (grep exists $part{$_}, split //, $compartment)[0];
    $sum += $PRIORITY{$common};
}
say $sum;

__DATA__
vJrwpWtwJgWrhcsFMMfFFhFp
jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL
PmmdzqPrVvPwwTWBwg
wMqvLMZHhHMvwLHjbvcjnnSBnvTQFn
ttgJtRGJQctTZtZT
CrZsJsPPZsGzwwsLwLmpwMDw

File: 03a.py
#!/usr/bin/python3
import sys

priority = dict([(chr(c), 1 + c - ord('a'))
                 for c in range(ord('a'), 1 + ord('z'))]
                + [(chr(c), 27 + c - ord('A'))
                   for c in range(ord('A'), 1 + ord('Z'))])

total = 0
with open(sys.argv[1], 'r') as fh:
    for line in fh:
        line = line.rstrip('\n')
        length = len(line)
        compartment = set(line[0:length // 2])
        common = compartment.intersection(set(line[length // 2:]))
        total += priority[list(common)[0]]
print(total)
